.github/workflows/main.yaml

```diff
@@ -0,0 +1,114 @@
+name: CI
+on:
+  push:
+    branches:
+      - 'main'
+      - 'scratch/**'
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+
+env:
+  BBI_VERSION: v3.3.0
+
+jobs:
+  check:
+    runs-on: ${{ matrix.config.os }}
+    name: ${{ matrix.config.os }} R ${{ matrix.config.r }} ${{ matrix.config.label }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - os: ubuntu-20.04
+            r: 4.2.3
+          - os: ubuntu-20.04
+            r: 4.3.1
+          - os: ubuntu-latest
+            r: release
+          - label: oldest
+            os: ubuntu-20.04
+            r: 4.0.5
+            bbi_version: v3.0.2
+            mpn: 2023-01-25
+            cran_override: 'https://mpn.metworx.com/snapshots/stable/2023-01-25'
+            rspm: 'https://packagemanager.posit.co/cran/__linux__/focal/2023-01-25'
+    env:
+      R_KEEP_PKG_SOURCE: yes
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install bbi
+        uses: metrumresearchgroup/actions/setup-bbi@v1
+        with:
+          version: ${{ matrix.config.bbi_version || env.BBI_VERSION }}
+      - uses: metrumresearchgroup/actions/mpn-latest@v1
+        if: matrix.config.label != 'oldest'
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: ${{ matrix.config.r }}
+          use-public-rspm: true
+          extra-repositories: 'https://mpn.metworx.com/snapshots/stable/${{ matrix.config.mpn || env.MPN_LATEST }}'
+        env:
+          RSPM: ${{ matrix.config.rspm }}
+          CRAN: ${{ matrix.config.cran_override }}
+      - name: Check out mrgmisc
+        if: matrix.config.label == 'oldest'
+        shell: bash
+        run: |
+          version=0.1.0
+          tdir=$(mktemp -d)
+          git clone --branch "$version" --depth 1 \
+            https://github.com/metrumresearchgroup/mrgmisc.git "$tdir"
+          echo "MRGMISC_PKG=local::$tdir" >>$GITHUB_ENV
+      - name: Check out nmrec
+        if: matrix.config.label == 'oldest'
+        shell: bash
+        run: |
+          # Even on oldest, install new enough nmrec for vignettes to
+          # build.
+          version=0.4.0
+          tdir=$(mktemp -d)
+          git clone --branch "$version" --depth 1 \
+            https://github.com/metrumresearchgroup/nmrec.git "$tdir"
+          # Delete testthat version constraint.  It doesn't matter in
+          # this context and causes pak's solver to abort.
+          sed -i 's/testthat (.*/testthat,/' "$tdir/DESCRIPTION"
```

Comment:
Can you explain this chunk? Why it was needed and what it's doing? Are you removing the testthat constraint from `nmrec`?

Reply:
(Saw this is outdated from your previous commit)

Reply:
> Can you explain this chunk? Why it was needed and what it's doing?

pak's dependency solver was aborting given nmrec's `testthat (>= 3.0.0)` constraint.  In other words, the `r-lib/actions/setup-r-dependencies@v2` step would fail.  I'll see if I can dig up the exact failure, though we may be past the retention period for the logs.

> Are you removing the testthat constraint from `nmrec`?

Yes.  As the comment mentions, it doesn't matter in this context.  That's a constraint for nmrec's tests, and we aren't running those.

Reply:
> I'll see if I can dig up the exact failure, though we may be past the retention period for the logs.

Yes, logs from those periods are expired already.

But I think it essentially just comes down to:

 * we were setting repos to be from the 2020-06 period (via MPN and RSPM)
 * testthat 3.0.0 isn't available until 2020-10

Given the above, pak was rightly failing saying it couldn't fulfill nmrec's requirement for testthat 3.0.0 or above.